src/main/java/com/mraof/minestuck/tileentity/redstone/SummonerTileEntity.java

```diff
@@ -108,12 +100,23 @@ public EntityType<?> getSummonedEntity()
 		return this.summonType;
 	}
 	
+	public void setSummonRange(int rangeIn)
+	{
+		this.summonRange = rangeIn;
+	}
+	
+	public int getSummonRange()
+	{
+		return MathHelper.clamp(this.summonRange, 1, 64);
+	}
+	
 	@Override
 	public void load(BlockState state, CompoundNBT compound)
 	{
 		super.load(state, compound);
 		
 		cooldownTimer = compound.getInt("cooldownTimer");
+		summonRange = compound.getInt("summonRange");
```

Comment:
Because "summonRange" didn't exist before, I suggest adding a `if(compound.contains("summonRange", Constants.NBT.TAG_ANY_NUMERIC))` check, so that the range is only read if it is in nbt. (`compound.getInt()` will default to 0 if the range tag is not there, which is not the default you want)

Reply:
Since it would also be optimal not to allow even players with commands to set the block to 0, perhaps I could create a check in getSummonRange() to convert any 0 values to the default? If not, I could make sure that getSummonRange is used anyways as it has a clamp. This would cover both issues with both older mod versions and attempts to set "illegal" bounds

Reply:
Putting it in `getSummonRange()` does work, but as an alternative I'd like to suggest putting it in `setSummonRange()`, and using it when reading from nbt. `this.setSummonRange(compound.getInt("summonRange"));`
If you do want to keep the clamp in the getter and not move it to the setter, you must make sure that you always use the getter to get the range, and never access `summonRange` directly like you currently do in `summonEntity()`.
Either way, the range would then instead end up at 1 when loaded for an old version. It works, but it could be better. I'd still suggest that you have a check for in the load function. In addition, to match the range before this PR, the default here could be 16 instead of 8.
```java
if(compound.contains("summonRange", Constants.NBT.TAG_ANY_NUMERIC))
	this.setSummonRange(compound.getInt("summonRange"));
else this.setSummonRange(16);
```